test/controllers/drivers_controller_test.rb

```diff
@@ -108,68 +144,105 @@
       # Ensure there is an existing driver saved
       # Assign the existing driver's id to a local variable
       # Set up the form data
-
+      driver1 = Driver.create(name: "Driver 100", vin: "123")
+      driver_id = driver1.id
+      driver_hash = {
+        driver: {
+          name: "Test",
+          vin: "5",
+        },
+      }
       # Act-Assert
       # Ensure that there is no change in Driver.count
-
+      expect {
+        patch driver_path(driver1.id), params: driver_hash
+        }.must_differ 'Driver.count', 0
       # Assert
       # Use the local variable of an existing driver's id to find the driver again, and check that its attributes are updated
       # Check that the controller redirected the user
-
+      must_redirect_to driver_path
+      edited_driver = Driver.find_by(id: driver_id)
+      expect(edited_driver.name).must_equal driver_hash[:driver][:name]
+      expect(edited_driver.vin).must_equal driver_hash[:driver][:vin]
     end
 
-    it "does not update any driver if given an invalid id, and responds with a 404" do
+    it "does not update any driver if given an invalid id, and redirects" do
       # Arrange
       # Ensure there is an invalid id that points to no driver
       # Set up the form data
-
+      nonexistant_driver = Driver.find_by(id: -1)
+      expect(nonexistant_driver).must_be_nil
+      driver_hash = {
+        driver: {
+          name: "Test",
+          vin: "5",
+        },
+      }
       # Act-Assert
       # Ensure that there is no change in Driver.count
-
+      expect{
+        patch driver_path(-1), params: driver_hash
+      }.must_differ 'Driver.count', 0
       # Assert
-      # Check that the controller gave back a 404
-
+      # Check that the controller redirects
+      must_redirect_to drivers_path
     end
 
-    it "does not create a driver if the form data violates Driver validations, and responds with a redirect" do
+    it "does not create a driver if the form data violates Driver validations, and responds with not_found" do
       # Note: This will not pass until ActiveRecord Validations lesson
       # Arrange
       # Ensure there is an existing driver saved
       # Assign the existing driver's id to a local variable
       # Set up the form data so that it violates Driver validations
+      driver1 = Driver.create(name: "Driver 100", vin: "123")
+      driver_hash = {
+        driver: {
+          name: "",
+          vin: "",
+        },
+      }
 
       # Act-Assert
       # Ensure that there is no change in Driver.count
+      expect{
+        patch driver_path(driver1.id), params: driver_hash
+      }.must_differ 'Driver.count', 0
 
       # Assert
-      # Check that the controller redirects
-
+      # Check that the controller responds with not_found
+      must_respond_with :not_found
```

Comment:
Again not found really doesn't apply here, you found it, it just violated validations.